wpimath/src/test/native/cpp/geometry/Translation2dTest.cpp

```diff
@@ -94,6 +94,42 @@ TEST(Translation2dTest, PolarConstructor) {
   EXPECT_DOUBLE_EQ(std::sqrt(3.0), two.Y().value());
 }
 
+TEST(Translation2dTest, Nearest) {
+  const Translation2d origin{0_m, 0_m};
+
+  const Translation2d translation1{
+      1_m, Rotation2d{units::degree_t{static_cast<double>(std::rand() % 360)}}};
+  const Translation2d translation2{
+      2_m, Rotation2d{units::degree_t{static_cast<double>(std::rand() % 360)}}};
+  const Translation2d translation3{
+      3_m, Rotation2d{units::degree_t{static_cast<double>(std::rand() % 360)}}};
+  const Translation2d translation4{
+      4_m, Rotation2d{units::degree_t{static_cast<double>(std::rand() % 360)}}};
+  const Translation2d translation5{
+      5_m, Rotation2d{units::degree_t{static_cast<double>(std::rand() % 360)}}};
```

Comment:
Why is the rotation randomized? It would be better if the test was deterministic.